src/Phinx/Config/Config.php

```diff
@@ -333,6 +334,60 @@ public function isVersionOrderCreationTime()
         return $versionOrder == self::VERSION_ORDER_CREATION_TIME;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getAvailableAdapters()
+    {
+        if (isset($this->values, $this->values['databases'])) {
+            $adapter_factory = AdapterFactory::instance();
+            $default_env = $this->getDefaultEnvironment();
+            $default_database = $this->getEnvironment($default_env);
+            $adapter_list = [];
+            foreach ($this->values['databases'] as $key => $options) {
+                $options = array_merge($default_database, $options);
+                $adapter = $adapter_factory->getAdapter($options['adapter'], $options);
+
+                if (isset($options['wrapper'])) {
+                    $adapter = $adapter_factory->getWrapper($options['wrapper'], $adapter);
+                }
+
+                // Use the TablePrefixAdapter if table prefix/suffixes are in use
+                if ($adapter->hasOption('table_prefix') || $adapter->hasOption('table_suffix')) {
+                    $adapter = $adapter_factory->getWrapper('prefix', $adapter);
+                }
+
+                $adapter_list[$key] = $adapter;
+            }
+
+            return $adapter_list;
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAvailableAdapter($name)
+    {
+        $adapters = $this->getAvailableAdapters();
```

Comment:
In theory no, as they are persistent per configuration. So yes, caching seems to be an good idea!